internal/handler/cmd_query.go

```diff
@@ -22,15 +22,45 @@ import (
 	"github.com/FerretDB/FerretDB/internal/handler/common"
 	"github.com/FerretDB/FerretDB/internal/handler/handlererrors"
 	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 // CmdQuery implements deprecated OP_QUERY message handling.
 func (h *Handler) CmdQuery(ctx context.Context, query *wire.OpQuery) (*wire.OpReply, error) {
-	cmd := query.Query().Command()
+	q := query.Query()
+	cmd := q.Command()
 	collection := query.FullCollectionName
 
+	v, _ := q.Get("speculativeAuthenticate")
+	if v != nil && (cmd == "ismaster" || cmd == "isMaster" || cmd == "hello") {
```

Comment:
```suggestion
	if v != nil && (cmd == "ismaster" || cmd == "isMaster") {
```

I believe @AlekSi is correct, the `hello` command is never sent in an OP_QUERY so we can probably remove it from the check.

Reply:
Removed `hello`

Reply:
It still works with the removal of `hello`.

Reply:
Thank a lot :pray:

Please resolve this comment if it's already addressed :hugs: